### PR TITLE
Allow the hiding of diffs of cloginrc

### PR DESCRIPTION
**Pull request by r4v5**
Some people put passwords into cloginrc. This PR adds the facility to hide diffs of that particular file for those people.

### CODE REVIEW

**Comment by r4v5** [comment]
Apparently this would cost you 3.1.0 compatibility.

**Comment by ghoneycutt** [comment]
This module has a dependency on stdlib 3.2.0 which contains `validate_bool()` so no issues there.

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -9,11 +9,40 @@
     end
 
     it { should compile.with_all_deps }
+
+    context 'with show_cloginrc_diff unset' do
+      it {should contain_file('rancid_cloginrc').with_show_diff('true')}
```
Comment (ghoneycutt) [inline spec/classes/init_spec.rb]: these should be booleans and not strings

**Comment by ghoneycutt** [comment]
Looks like show_diff was added in v3.6.0.. how odd.

**Comment by r4v5** [comment]
show_diff as a file param seems to exist as far back as 3.2.1: https://docs.puppetlabs.com/references/3.2.1/type.html#file but wasn't there in 3.1.

**Comment by ghoneycutt** [comment]
+1 

Given this module is pre 1.0.0

**Comment by eoly** [comment]
Merged #15 which covers the changes here.

**Pull request closed without merging**